doc/DataPlaneCodeGeneration/Subclients.md

```diff
@@ -0,0 +1,295 @@
+    {
+-       protected AccountsClient(){}
++       protected Accounts(){}
+-       public AccountsClient(string endpoint, TokenCredential credential, PurviewAccountClientOptions options = null){}
++       internal Accounts(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, TokenCredential tokenCredential, string endpoint, string apiVersion)
+    }
+}
+
+//subclient: Collections Generated\Collections.cs
+namespace Azure.Analytics.Purview.Account
+{
+-   public partial class CollectionsClient
++   public partial class Collections
+    {
+-       protected CollectionsClient()
++       protected Collections()
+-       public CollectionsClient(string endpoint, TokenCredential credential, PurviewAccountClientOptions options = null){}
++       internal Collections(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, TokenCredential tokenCredential, string endpoint, string apiVersion) {}
+    }
+}
+
+//subclient: ResourceSetRules Generated\ResourceSetRules.cs
+namespace Azure.Analytics.Purview.Account
+{
+-   public partial class ResourceSetRulesClient
++   public partial class ResourceSetRules
+    {
+-       protected ResourceSetRulesClient()
++       protected ResourceSetRules()
+-       public ResourceSetRulesClient(string endpoint, TokenCredential credential, PurviewAccountClientOptions options = null){}
++       internal ResourceSetRules(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, TokenCredential tokenCredential, string endpoint, string apiVersion) {}
+    }
+}
+
+```
+
+</details>
+
+### Client hierarchy
+
+Each client can be made a subclient of another client (without circular dependency) using CodeGenClientAttribute.ParentClient parameter.
+Since CodeGenClientAttribute is applied to the client type, user will explicitly specify new client type name. According to [guideline](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-subclients), type should have a name without "Client" suffix and an internal constructor. A factory method will be added to the type, specified in ParentClient.
+
+<details>
+
+**Generated code before:**
+
+``` C#
+//Generated\AccountsClient.cs
+namespace Azure.Analytics.Purview.Account
+{
+    public partial class AccountsClient
+    {
+        public AccountsClient(string endpoint, TokenCredential credential, PurviewAccountClientOptions options = null){}
+    }
+}
+
+//Generated\CollectionsClient.cs
+namespace Azure.Analytics.Purview.Account
+{
+    public partial class CollectionsClient
+    {
+        public CollectionsClient(string endpoint, TokenCredential credential, PurviewAccountClientOptions options = null){}
+    }
+}
+
+//Generated\ResourceSetRulesClient.cs
+namespace Azure.Analytics.Purview.Account
+{
+    public partial class ResourceSetRulesClient
+    {
+        public ResourceSetRulesClient(string endpoint, TokenCredential credential, PurviewAccountClientOptions options = null){}
+    }
+}
+```
+
+**Add client hierarchy customization (Customizations.cs):**
+
```

Comment:
Curious about the steps here. Do i need to first generate the original code, add Customizations.cs into the source code folder and generate again?

Reply:
> Curious about the steps here. Do i need to first generate the original code, add Customizations.cs into the source code folder and generate again?

Without first generating the original code, you will not know which clients are in the PR and write the Customization.cs to setup the hierarchy. If you can figure out the clients and set the client hierarchy, you will not generate the original code.